clippy_lints/src/unnecessary_string_new.rs

```diff
@@ -0,0 +1,58 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::{match_def_path, paths};
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Detects cases of `&String::new()` being passed as an argument to a function expecting `&str`
+    ///
+    /// ### Why is this bad?
+    ///
+    /// `&String::new()` is longer and may be confusing
+    ///
+    /// ### Example
+    /// ```rust
+    /// vec!["1", "2", "3"].join(&String::new());
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// vec!["1", "2", "3"].join("");
+    /// ```
+    #[clippy::version = "1.61.0"]
+    pub UNNECESSARY_STRING_NEW,
+    style,
+    "detects cases of `&String::new()` being passed as an argument to a function expecting `&str`"
+}
+declare_lint_pass!(UnnecessaryStringNew => [UNNECESSARY_STRING_NEW]);
+
+impl<'tcx> LateLintPass<'tcx> for UnnecessaryStringNew {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>) {
+        let applicability = Applicability::MachineApplicable;
+        if_chain! {
+            if let ExprKind::AddrOf(.., inner_expr) = expr.kind;
```

Comment:
As mentioned on Zulip, I would also check for the mutability here and that it is a `Ref`:
```suggestion
            if let ExprKind::AddrOf(BorrowKind::Ref, Mutability::Not, inner_expr) = expr.kind;
```